### ISSUE TITLE
Script with syntax error broke game

### ISSUE THREAD

**Issue by Brilliand** [danielyxie/bitburner#1390]
I wrote a script (designed to automatically scan for and nuke servers), and tried to run it.  A message popped up saying that I had a syntax error.  I went looking for the syntax error, couldn't find it, so I went back and tried to run the script again to see the error again... that's when I discovered that...

1) As far as the "run" command was concerned, that script was already running and could not be started again.
2) As for as the "kill", "killall" and "free" commands were concerned, the script was not running and thus there was nothing to kill.
3) The "mv" command also considered the script to be already running.

I manually saved and refreshed the page, hoping that that would cause the game to correct itself... and that set me way back, to a gamestate that looked _sort of_ like the beginning of the game, except that I still had some of the things from before (such as my money supply).  Notably, I still had my factions from before, but I was _also_ getting offers to join those factions again, resulting in multiple copies of the faction on the factions list.

**Comment by Brilliand** [comment]
The offending line in the script I wrote was:

`to_scan = to_scan.concat(scan(target).filter(a=>!to_scan.includes(a)));`

Now that I'm testing again, the "kill" command is reporting that it killed the script successfully, so I may have misremembered 2). Nonetheless, no matter how many times I kill the script, the "run" command maintains that it's already running.

**Comment by JanisBoegershausen** [comment]
Hey, I have experienced the same issue twice just now and have done some testing. 

As described, once a faulty script "runs" (even though it can't really run due to the syntax error), it can't be killed, does not show up in the "Active Scripts"-tab, and can't be rerun (because it's already running). From my experience, it only seems to occur if the error is a syntax error. 
The "Force kill all active scripts" button in the options fixes this but the problem is, that from what it seems like, saving with such a script still "running", **corrupts the save file**. 

### The easiest way to recreate is: 
1) Create a new game
2) Skip the tutorial
3) Write a new script with invalid syntax (for example just a script containing "class" and nothing else)
4) Run the script, which will result in "SyntaxError: Unexpected token (1:5)"
5) Save and reload the page
Result: Tutorial shows up, some things are reset, others are not, and the server is now "home-0" instead of just "home", meaning all files are lost. As @Brilliand mentioned, apparently the old "home" server still exists somewhere and I would love to know how to access it. 

### Workaround:
If this happens, do not reload the page! First of all, edit the script, so that all syntax errors are fixed, then head to the "Options" and click "Force kill all active scripts". Now save and reload. This should not result in any issues from what I have tested

**Comment by Brilliand** [comment]
To get access to the old "home" server, use the "connect" command to get access to one of the adjacent servers (for example, my first server was "n00dles" so I used "connect n00dles"), then from there connect to the server "home" as though it's an ordinary server (I used scan-analyze with AutoLink.exe available, but I'm pretty sure "connect home" would work just as well).

Note that "scan-analyze" from the new "home-0" server returns nothing, even after installing augmentations (which deletes the old "home" server).  However, running "scan" from the new "home-0" server reveals that the save corruption resulted in an entire duplicate server hierarchy being created, most of which can only be accessed by IP address because any attempt to access them by name switches you over to the old server hierarchy.
